mempool/src/tests/shared_mempool_test.rs

```diff
@@ -726,36 +726,33 @@ fn test_state_sync_events_committed_txns() {
     assert_eq!(timeline.get(0).unwrap().1, kept_txn);
 }
 
-// primary_peers = k, fallbacks > 0
 #[test]
-fn test_k_policy_broadcast_no_fallback() {
-    // all nodes
+fn test_fn_failover() {
```

Comment:
Should we add a test of what happens if there are no upstream or peers available?